File: fe/user_management/lib/features/list_users/data/models/user_model.dart
import 'package:user_management/features/list_users/domain/entities/user_entity.dart';

class UserModel extends UserEntity {
  const UserModel({required lastName, required firstName, required uuidString})
      : super(lastName: lastName, firstName: firstName, uuidString: uuidString);

  factory UserModel.fromJson(Map<String, dynamic> jsonMap) {
    return UserModel(
        firstName: jsonMap['firstName'],
        lastName: jsonMap['lastName'],
        uuidString: jsonMap['uuid']);
  }

  Map<String, dynamic> toJson() {
    return {
      "lastName": lastName,
      "firstName": firstName,
      "uuid": uuidString
    };
  }
}

File: fe/kotlin/UserManagement/app/src/main/java/com/example/usermanagement/features/get_users/data/models/UserModel.kt
package com.example.usermanagement.features.get_users.data.models

import com.example.usermanagement.features.get_users.domain.entities.UserEntity
import com.google.gson.Gson
import java.util.Objects
import java.util.UUID

class UserModel(uuid: UUID, lastName: String, firstName: String)
    : UserEntity(uuid, lastName, firstName) {
    companion object {
        fun fromJSON(jsonString: String): UserModel {
            return Gson().fromJson(jsonString, UserModel::class.java)
        }
    }

    override fun equals(other: Any?): Boolean {
        return if (other is UserModel) {
            other.uuid == uuid && other.lastName == lastName && other.firstName == firstName
        } else false
    }

    override fun hashCode(): Int {
        return Objects.hash(uuid, lastName, firstName)
    }

    fun toJson(): String {
        return Gson().toJson(this)
    }
}